### PR TITLE
ELPP-3614 Add form-field-info-link pattern

### PR DESCRIPTION
**Pull request by davidcmoulton**
Implements https://github.com/elifesciences/pattern-library/pull/868.

### CODE REVIEW

File: src/ViewModel/EmailCta.php [hunk 1]
```diff
@@ -18,23 +18,28 @@
...
     private $subHeader;
     private $compactForm;
+    private $formFieldInfoLink;
 
     public function __construct(
         string $headerText,
         string $subHeader,
-        CompactForm $compactForm
+        CompactForm $compactForm,
+        FormFieldInfoLink $formFieldInfoLink
```
Comment (thewilkybarkid) [inline src/ViewModel/EmailCta.php]: Should be optional?

File: src/ViewModel/FormFieldInfoLink.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        Assertion::notBlank($url);
+
+        $this->text = $text;
+        $this->url = $url;
+        if ($alignLeft) {
+            $this->alignLeft = $alignLeft;
+        }
+    }
+
+    public static function alignedLeft(string $text, string $url)
```
Comment (thewilkybarkid) [inline src/ViewModel/FormFieldInfoLink.php]: These need return types.

File: src/ViewModel/FormFieldInfoLink.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public static function alignedRight(string $text, string $url)
+    {
+        return new static($text, $url);
+    }
+
+    public function getLocalStyleSheets() : Traversable
+    {
+        yield 'resources/assets/css/form-field-info-link.scss';
```
Comment (thewilkybarkid) [inline src/ViewModel/FormFieldInfoLink.php]: `.css`

File: src/ViewModel/TextField.php [hunk 4]
```diff
@@ -62,12 +64,14 @@ protected function __construct(
...
         $this->state = $state;
         $this->messageGroup = $messageGroup;
+        $this->formFieldInfoLink = $formFieldInfoLink;
     }
 
     public static function emailInput(
         FormLabel $label,
         string $id,
         string $name,
+        FormFieldInfoLink $formFieldInfoLink,
```
Comment (thewilkybarkid) [inline src/ViewModel/TextField.php]: Should be optional?

Comment (davidcmoulton) [reply src/ViewModel/TextField.php]: We'll always need a privacy notice link by an email field, hence mandating the link here. The tradeoff is that the honeypot has to have it, even though it's meaningless in that context.

Should the mandatory nature of the field be imposed at a higher level, i.e. somewhere in `journal`?

Comment (thewilkybarkid) [reply src/ViewModel/TextField.php]: We should let the implementor handle it (since it's not required by the pattern).

File: bin/update [hunk 5]
```diff
@@ -32,7 +32,7 @@ $output = function (string $type, string $buffer) {
     }
 };
 
-$execute = function(array $command, $timeout = 60) use ($output) {
+$execute = function (array $command, $timeout = 60) use ($output) {
```
Comment (thewilkybarkid) [inline bin/update]: Should be reverted. (While correct, unrelated to this PR.)

File: src/ViewModel/EmailCta.php [hunk 6]
```diff
@@ -18,23 +18,27 @@
...
         $this->headerText = $headerText;
         $this->subHeader = $subHeader;
         $this->compactForm = $compactForm;
+        $this->formFieldInfoLink = $formFieldInfoLink;
     }
 
     public function getLocalStyleSheets() : Traversable
     {
         yield 'resources/assets/css/email-cta.css';
+        yield 'resources/assets/css/form-field-info-link.css';
```
Comment (thewilkybarkid) [inline src/ViewModel/EmailCta.php]: Not needed, entry in `getComposedViewModels()` will add it when necessary.

File: src/ViewModel/TextField.php [hunk 7]
```diff
@@ -162,6 +168,7 @@ public function getTemplateName() : string
     public function getStyleSheets() : Traversable
     {
         yield 'resources/assets/css/text-fields.css';
+        yield 'resources/assets/css/form-field-info-link.css';
```
Comment (thewilkybarkid) [inline src/ViewModel/TextField.php]: Need to replace `SimplifyAssets` with `ComposedAssets` instead of this.

**Comment by davidcmoulton** [comment]
@thewilkybarkid Makes sense 👍

**Pull request merged**